x/supplier/keeper/msg_server_stake_supplier.go

```diff
@@ -140,15 +125,20 @@ func (k msgServer) StakeSupplier(ctx context.Context, msg *suppliertypes.MsgStak
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	// Send the coins from the message signer account to the staked supplier pool
 	supplierStakingFee := k.GetParams(ctx).StakingFee
-	stakeWithFee := sdk.NewCoins(coinsToEscrow.Add(*supplierStakingFee))
-	err = k.bankKeeper.SendCoinsFromAccountToModule(ctx, msgSignerAddress, suppliertypes.ModuleName, stakeWithFee)
-	if err != nil {
-		logger.Info(fmt.Sprintf("ERROR: could not send %v coins from %q to %q module account due to %v", coinsToEscrow, msgSignerAddress, suppliertypes.ModuleName, err))
-		return nil, status.Error(codes.InvalidArgument, err.Error())
+
+	if err = k.transferSupplierStakeDiff(ctx, msgSignerAddress, supplierCurrentStake, *msg.Stake); err != nil {
+		logger.Info(fmt.Sprintf("ERROR: could not transfer supplier stake difference due to %v", err))
```

Comment:
1. Internal errors should ALWAYS be logged at the error level.
2. We don't need to prefix the log with the level, at least when it matches the level used (i.e. we use `info` for client-side errors).
3. We should ALWAYS use `%s` to interpolate errors into strings unless there's a specific reason to do otherwise.  `%v` will result in some undesirable output which includes the path to the errors.go file, and `%+v` will include the stack trace.

```suggestion
		logger.Error(fmt.Sprintf("could not transfer supplier stake difference due to %s", err))
```